### PR TITLE
cacher.go: embed storage.Interface into cacher

### PR DESCRIPTION
**Pull request by xiang90**
Continuous effort to simplify cacher implementation.

### CODE REVIEW

**Comment by xiang90** [comment]
@timothysc Can you please take a look?

**Comment by timothysc** [comment]
Sorry, was on PTO... catching up

**Comment by timothysc** [comment]
much cleaner, thx. 
LGTM.

**Comment by wojtek-t** [comment]
It seems that this PR completely broke all the performance tests. 

@k8s-oncall

**Comment by wojtek-t** [comment]
I'm going to revert it to fix the tests and unblock merge-queue.

File: pkg/storage/cacher.go [hunk 1]
```diff
@@ -306,26 +286,11 @@ func (c *Cacher) Watch(ctx context.Context, key string, resourceVersion string,
 }
 
 // Implements storage.Interface.
-func (c *Cacher) WatchList(ctx context.Context, key string, resourceVersion string, filter Filter) (watch.Interface, error) {
```
Comment (xiang90) [inline pkg/storage/cacher.go]: @wojtek-t can you try to fix this one instead of reverting the whole thing?

Comment (wojtek-t) [inline pkg/storage/cacher.go]: Sorry - I'm super busy today and I'm taking half a day off and I don't have time to really confirm it.

And we were already blocked for almost 10 hours, and we need to unblock merge-queue ASAP.

Comment (xiang90) [inline pkg/storage/cacher.go]: no problem. i totally understand. i will try to send a pr to fix this later.

**Pull request merged**